handlers/upload.go

```diff
@@ -76,7 +77,46 @@ func (s Server) entryPut() http.HandlerFunc {
 			return
 		}
 
-		metadata, err := s.entryMetadataFromRequest(r)
+		// We allow changing filename, expiration, note, and optionally passphrase.
+		// Passphrase change is specified with either passphrase (string) to set/update,
+		// or passphraseAction:"clear" to remove.
+		type editPayload struct {
```

Comment:
This should happen in a standalone function. See guestLinkFromRequest for example.